utility_analysis_new/utility_analysis.py

```diff
@@ -26,19 +31,62 @@ class UtilityAnalysisOptions:
 
 
 def perform_utility_analysis(col,
+                             backend: pipeline_backend.PipelineBackend,
                              options: UtilityAnalysisOptions,
                              data_extractors: pipeline_dp.DataExtractors,
                              public_partitions=None):
     """Performs utility analysis for DP aggregations.
 
   Args:
     col: collection where all elements are of the same type.
+    backend: PipelineBackend with which the utility analysis will be run.
     options: options for utility analysis.
     data_extractors: functions that extract needed pieces of information
           from elements of 'col'.
     public_partitions: A collection of partition keys that will be present
           in the result. If not provided, the utility analysis with private
           partition selection will be performed.
   """
-    raise NotImplementedError(
-        "perform_utility_analysis is not implemented yet.")
+    budget_accountant = pipeline_dp.NaiveBudgetAccountant(
+        total_epsilon=options.eps, total_delta=options.delta)
+    engine = dp_engine.UtilityAnalysisEngine(
+        budget_accountant=budget_accountant, backend=backend)
+    result = engine.aggregate(col,
+                              params=options.aggregate_params,
+                              data_extractors=data_extractors,
+                              public_partitions=public_partitions)
+    budget_accountant.compute_budgets()
+
+    def create_aggregate_error_compound_combiner(
```

Comment:
Nit: this function is large enough (it will be bigger later), it's better to extract it as a free function in this module (with _ in the beginning of the name)